src/promql/src/planner.rs

```diff
@@ -431,10 +441,78 @@ impl PromPlanner {
         )));
 
         // make table scan with filter exprs
-        let table_scan = self
+        let mut table_scan = self
             .create_table_scan_plan(&table_name, filters.clone())
             .await?;
 
+        // make a projection plan if there is any `__value__` matcher
+        if let Some(value_matchers) = &self.ctx.value_column_matcher {
+            let col_set = self.ctx.value_columns.iter().collect::<HashSet<_>>();
+            // opt-in set
+            let mut result_set = HashSet::new();
+            // opt-out set
+            let mut reverse_set = HashSet::new();
+            for matcher in value_matchers {
+                match &matcher.op {
+                    MatchOp::Equal => {
+                        if col_set.contains(&matcher.value) {
+                            result_set.insert(matcher.value.clone());
+                        } else {
+                            return Err(ColumnNotFoundSnafu {
+                                col: self.ctx.table_name.clone().unwrap(),
+                            }
+                            .build());
+                        }
+                    }
+                    MatchOp::NotEqual => {
+                        if col_set.contains(&matcher.value) {
+                            reverse_set.insert(matcher.value.clone());
+                        } else {
+                            return Err(ValueNotFoundSnafu {
+                                table: self.ctx.table_name.clone().unwrap(),
+                            }
+                            .build());
+                        }
+                    }
```

Comment:
for `Equal` and `NotEqual`, which is the behavior of Prometheus, empty result or error?

Reply:
PromQL doesn't throw error in most cases like wrong table name, wrong column name etc. But we need to report error for better UE. I plan to introduce a config like "prom_compatible_mode" to switch this behavior. But in this case `__field__` doesn't exist in promql. It's non-"prom_compatible_mode" only feature so I just throw an error.

Reply:
What about `Re` and `NotRe`, Is there a chance to throw error? It's like `Equal` and `Re` behaves differently.

Reply:
They are not accurate matches, so I ignore the unmatching case for regex.